modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C

```diff
@@ -8,68 +8,41 @@
 {
-  InputParameters params = validParams<RadialReturnStressUpdate>();
-  params.addClassDescription("This class uses the discrete material in a radial return isotropic "
-                             "power law creep model.  This class can be used in conjunction with "
-                             "other creep and plasticity materials for more complex simulations.");
+  InputParameters params = validParams<RadialReturnCreepStressUpdateBase>();
+  params.addClassDescription(
+      "This class uses the stress update material in a radial return isotropic power law creep "
+      "model.  This class can be used in conjunction with other creep and plasticity materials for "
+      "more complex simulations.");
 
   // Linear strain hardening parameters
   params.addRequiredParam<Real>("coefficient", "Leading coefficent in power-law equation");
   params.addRequiredParam<Real>("n_exponent", "Exponent on effective stress in power-law equation");
   params.addParam<Real>("m_exponent", 0.0, "Exponent on time in power-law equation");
   params.addRequiredParam<Real>("activation_energy", "Activation energy");
   params.addParam<Real>("gas_constant", 8.3143, "Universal gas constant");
-  params.addParam<Real>("start_time", 0.0, "Start time (if not zero)");
-  params.addCoupledVar("temperature", 0.0, "Coupled temperature");
-  params.addDeprecatedParam<std::string>(
-      "creep_prepend",
-      "",
-      "String that is prepended to the creep_strain Material Property",
-      "This has been replaced by the 'base_name' parameter");
-  params.set<std::string>("effective_inelastic_strain_name") = "effective_creep_strain";
+  params.addDeprecatedParam<Real>("start_time",
+                                  "Start time (if different than the executioner's start_time).",
+                                  "This is automatically set from by the executioner start_time");
 
   return params;
 }
 
 PowerLawCreepStressUpdate::PowerLawCreepStressUpdate(const InputParameters & parameters)
-  : RadialReturnStressUpdate(parameters),
-    _creep_prepend(getParam<std::string>("creep_prepend")),
+  : RadialReturnCreepStressUpdateBase(parameters),
     _coefficient(getParam<Real>("coefficient")),
     _n_exponent(getParam<Real>("n_exponent")),
     _m_exponent(getParam<Real>("m_exponent")),
     _activation_energy(getParam<Real>("activation_energy")),
     _gas_constant(getParam<Real>("gas_constant")),
-    _start_time(getParam<Real>("start_time")),
-    _has_temp(isCoupled("temperature")),
-    _temperature(_has_temp ? coupledValue("temperature") : _zero),
-    _creep_strain(declareProperty<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain")),
-    _creep_strain_old(
-        getMaterialPropertyOld<RankTwoTensor>(_base_name + _creep_prepend + "creep_strain"))
-{
-}
-
-void
-PowerLawCreepStressUpdate::initQpStatefulProperties()
+    _start_time(isParamValid("start_time") ? getParam<Real>("start_time") : _app.getStartTime())
```

Comment:
This works, except that if someone had a model that started at a nonzero time and did not specify start_time, it would run differently now.

Reply:
Yeah that's true...